frontend/src/components/RecentSearch.jsx

```diff
@@ -32,13 +33,13 @@ export default function RecentSearches({ recentSearches }) {
                 <div className='divider'/> 
               </div>
               {todaySearches.map((search, index) => (
-                <a
-                  href={`#search-${index}`}
-                  key={`today-${index}`}
-                  className="dropdown-item"
-                >
-                  <p>{search.name}</p>
-                </a>
+                 <Link
+                 to={`/search?query=${encodeURIComponent(search.name)}`}
```

Comment:
we already have the data and don't need to be doing all this stuff here. Just pass it the property in our saved search:

```js
to={search.url}
```